nixos/modules/services/web-servers/hydron.nix

```diff
@@ -0,0 +1,111 @@
+{ config, lib, pkgs, ... }:
+
+let cfg = config.services.hydron;
+in with lib; {
+  options.services.hydron = {
+    enable = mkEnableOption "hydron";
+
+    dataDir = mkOption {
+      type = types.path;
+      default = "/var/lib/hydron";
+      example = "/home/okina/hydron";
+      description = "Location where hydron runs and stores data.";
+    };
+
+    interval = mkOption {
+      type = types.string;
+      default = "hourly";
+      example = "06:00";
+      description = ''
+        How often we run hydron import and fetch_tags. Runs by default every hour.
+
+        The format is described in
+        <citerefentry><refentrytitle>systemd.time</refentrytitle>
+        <manvolnum>7</manvolnum></citerefentry>.
+      '';
+    };
+
+    listenAddress = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "127.0.0.1:8010";
+      description = "Listen on a specific IP address and port.";
+    };
+
+    importPaths = mkOption {
+      type = types.nonEmptyListOf types.path;
+      default = [ (cfg.dataDir + "/images") ];
+      example = [ "/home/okina/Pictures" ];
+      description = "Paths that hydron will recursively import.";
+    };
+
+    fetchTags = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Fetch tags for imported images and webm from gelbooru.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.hydron = {
+      description = "hydron";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      preStart = ''
+        # Ensure folder exists and permissions are correct
+        mkdir -p ${cfg.dataDir}/images ${concatStringsSep " " cfg.importPaths}
+        chmod 750 ${cfg.dataDir}
+        chown -R hydron:hydron ${cfg.dataDir}
+      '';
+
+      serviceConfig = {
+        PermissionsStartOnly = true;
+        User = "hydron";
+        Group = "hydron";
+        ExecStart = "${pkgs.hydron}/bin/hydron serve"
+        + optionalString (cfg.listenAddress != null) " -a ${cfg.listenAddress}";
+      };
+    };
+
+    systemd.services.hydron-fetch = {
+      description = "Import paths into hydron and fetch tags";
+      
+      script = ''
+        ${pkgs.hydron}/bin/hydron import ${concatStringsSep " " cfg.importPaths}
```

Comment:
Along with the `importPaths` suggestion, you can add `"${dataDir}/imports"` here. Also: You can use `lib.escapeShellArgs cfg.importPaths` here instead to handle the escaping. (`lib.escapeShellArg` *could* be used as well for the `mkdir` above, maybe a bit excessive though)